test/integration/backward_compatible/parallel/ListenersOnReconnectTest.js

```diff
@@ -16,66 +16,78 @@
 'use strict';
 
 const { expect } = require('chai');
-const RC = require('../RC');
-const { Client } = require('../../../');
-const TestUtil = require('../../TestUtil');
+const RC = require('../../RC');
+const TestUtil = require('../../../TestUtil');
 
 describe('ListenersOnReconnectTest', function () {
     let client;
     let cluster;
     let map;
 
+    const testFactory = new TestUtil.TestFactory();
+
     beforeEach(async function () {
-        cluster = await RC.createCluster(null, null);
+        cluster = await testFactory.createClusterForParallelTest();
     });
 
     afterEach(async function () {
-        await map.destroy();
-        await client.shutdown();
-        return RC.terminateCluster(cluster.id);
+        await testFactory.cleanUp();
     });
 
     async function closeTwoMembersOutOfThreeAndTestListener(done, isSmart, membersToClose, turnoffMember) {
-        const members = await Promise.all([
-            RC.startMember(cluster.id),
-            RC.startMember(cluster.id),
-            RC.startMember(cluster.id)
-        ]);
-        client = await Client.newHazelcastClient({
-            clusterName: cluster.id,
-            network: {
-                smartRouting: isSmart
-            },
-            properties: {
-                'hazelcast.client.heartbeat.interval': 1000,
-                'hazelcast.client.heartbeat.timeout': 3000
-            }
-        });
-        map = await client.getMap('testmap');
-
-        const listener = {
-            added: (entryEvent) => {
-                try {
-                    expect(entryEvent.name).to.equal('testmap');
-                    expect(entryEvent.key).to.equal('keyx');
-                    expect(entryEvent.value).to.equal('valx');
-                    expect(entryEvent.oldValue).to.be.equal(null);
-                    expect(entryEvent.mergingValue).to.be.equal(null);
-                    expect(entryEvent.member).to.not.be.equal(null);
-                    done();
-                } catch (err) {
-                    done(err);
+        let doneCalled = false;
```

Comment:
done should be called only once this is a requirement of mocha. I think the error message when done is called twice is not good, I don't remember. Instead of reverting this logic, we can throw error with a useful message in the place where we should not be. For example we can add an else block to the listener and throw an error `listener called twice`

Reply:
To me, both errors look the same. I would prefer the old version as the code is much simpler but I don't mind if you do what you described. We just need to make sure, we fail the test on multiple done calls

Reply:
ok does not matter, reverted